src/com/softwaret/kpdf/util/extension/ApplicationCallExtension.kt

```diff
@@ -1,16 +1,105 @@
 package com.softwaret.kpdf.util.extension
 
 import com.softwaret.kpdf.response.Response
-import io.ktor.application.ApplicationCall
-import io.ktor.auth.jwt.JWTPrincipal
-import io.ktor.auth.principal
-import io.ktor.response.respond
+import com.softwaret.kpdf.response.file.FileResponse
+import com.softwaret.kpdf.util.log.Log
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.auth.jwt.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.*
+import kotlin.coroutines.CoroutineContext
 
 val ApplicationCall.userLoginFromPrincipal
     get() = run {
         val login = principal<JWTPrincipal>()?.payload?.userLogin
         checkNotNull(login) { "User authentication required" }
     }
 
-suspend fun ApplicationCall.respondWith(response: Response) =
+suspend fun ApplicationCall.respondWithFile(
+    fileResponse: FileResponse,
+    fileDir: File,
+    context: CoroutineContext = Dispatchers.IO
+) = when (fileResponse.code) {
+    HttpStatusCode.NotFound -> respond(fileResponse.code)
+    else -> respondFile(fileDir, fileResponse.fileName, fileResponse.file, context)
+}
+
+private suspend fun ApplicationCall.respondFile(
+    fileDir: File,
+    fileName: String,
+    fileBytes: ByteArray,
+    context: CoroutineContext = Dispatchers.IO
+) {
+    try {
+        withContext(context) {
+            val file = File(fileDir, fileName)
+            val fileOutStream = FileOutputStream(file)
+            fileOutStream.write(fileBytes)
+            fileOutStream.close()
+            respondFile(fileDir, fileName)
+            file.delete()
```

Comment:
Dlaczego usuwasz plik do którego właśnie coś zapisałeś?

Reply:
Usuwam to co zostało zwrócone w odpowiedzi na to zapytanie